pcdsutils/profile.py

```diff
@@ -54,7 +55,7 @@ def profiler_context(
     use_global_profiler: bool = False,
     output_now: bool = True,
     min_threshold: float = 0,
-) -> LineProfiler:
+) -> Iterator[LineProfiler]:
```

Comment:
Maybe [`typing.ContextManager[LineProfiler]`](https://docs.python.org/3.8/library/typing.html#typing.ContextManager)?

Reply:
Haha every time I do this there's some more precise way to hint things, I'll look at this again

Reply:
The docs for this one are a whirlwind:
```
class typing.ContextManager(Generic[T_co])
A generic version of [contextlib.AbstractContextManager](https://docs.python.org/3/library/contextlib.html#contextlib.AbstractContextManager).

New in version 3.5.4.

New in version 3.6.0.

Deprecated since version 3.9: [contextlib.AbstractContextManager](https://docs.python.org/3/library/contextlib.html#contextlib.AbstractContextManager) now supports []
```

I think the cleanest thing to do is to use the 3.9 version and make sure the future import is on